virttest/utils_misc.py

```diff
@@ -2565,11 +2565,12 @@ def __get_unit_index(M):
     from_index = __get_unit_index(unit)
     to_index = __get_unit_index(order_magnitude)
     scale = int(factor) ** (to_index - from_index)
-    if scale > 0:
-        data_size = float(value) / abs(scale)
+    data_size = float(value) / scale
+    # Control precision to avoid scientific notaion
+    if data_size.is_integer():
```

Comment:
>>> data = float(1.232323)
>>> print "%.20f" % data
1.23232300000000005724

so the format output is not precise.
Maybe we should consider adopt decimal module here?